### ISSUE TITLE
\Magento\Framework\Filesystem\Io\File::read() method is not compatible with \Magento\Framework\Filesystem\Io\IoInterface::read()

### ISSUE THREAD

**Issue by bnymn** [magento/magento2#27866]
Let's first define some classes here that I am going to mention a lot in the description.
```
use Magento\Framework\Filesystem\Io\IoInterface;
use Magento\Framework\Filesystem\Io\Sftp;
use Magento\Framework\Filesystem\Io\File;
```

I am using `Sftp` class in order to download some files from a remote server. In my class, I depend on `IoInterface` and resolve the dependency in `di.xml`. In my test, I would like to use `File` implementation, because I have more control over the local file system then the remote SFTP server.

In the method definition of `IoInterface`, it says that the second argument can be a resource.
![image](https://user-images.githubusercontent.com/3506402/79351820-6bffa180-7f39-11ea-9821-a8e0087b0100.png)
 
If I give the second argument as a resource it works just fine in the `Sftp` implementation. However, the `File` implementation uses `copy` function of PHP, and it expects the second argument to be a string **only**.
![image](https://user-images.githubusercontent.com/3506402/79352477-34ddc000-7f3a-11ea-93f4-4965f22b7c53.png)

### Preconditions (*)

1. Magento 2.4-develop
2. PHP 7.4

### Steps to reproduce (*)

1. Try to run the following code block in a fresh magento installation.
```php
$file = \Magento\Framework\App\ObjectManager::getInstance(\Magento\Framework\Filesystem\Io\File::class);
file_put_contents('/tmp/sample_file', 'Some test file');
$targetHandle = fopen('/tmp/' . rand(), 'w');
$file->cd('/tmp');
$result = $file->read('sample_file', $targetHandle);
var_dump($result);
```

### Expected result (*)

1. I expect the file to be copied to the target path.

### Actual result (*)

1. Unfortunately, it gives an error because the copy function requires the second argument to be string.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Delta 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-37905` were created

**Issue Available**: @engcom-Delta, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @bnymn. Thank you for your report. 
The issue has been fixed in magento/magento2#27869 by @bnymn in 2.4-develop branch
Related commit(s):
 - [cbc663739b5a3e2f062e1e87d96a073deff84767](https://github.com/magento/magento2/commit/cbc663739b5a3e2f062e1e87d96a073deff84767)
 - [d177789b991704da08996f42ef00ff886baa982a](https://github.com/magento/magento2/commit/d177789b991704da08996f42ef00ff886baa982a)
 - [40243209277f540cb376e7d952a98a967c00546c](https://github.com/magento/magento2/commit/40243209277f540cb376e7d952a98a967c00546c)
 - [c160b44e85a7516a5fdda8217f685b09c57acfe5](https://github.com/magento/magento2/commit/c160b44e85a7516a5fdda8217f685b09c57acfe5)
 - [ed111ae759aa7b924846f9ded51030a25be0082f](https://github.com/magento/magento2/commit/ed111ae759aa7b924846f9ded51030a25be0082f)
 - [a00f2d0d86da112513a6d23c77fbe9c830d32272](https://github.com/magento/magento2/commit/a00f2d0d86da112513a6d23c77fbe9c830d32272)
 - [b480c4d91897864701deaa044ac3f9958d7d090b](https://github.com/magento/magento2/commit/b480c4d91897864701deaa044ac3f9958d7d090b)
 - [2b2aceed74fceefbcafdd90dda3a8438a00328eb](https://github.com/magento/magento2/commit/2b2aceed74fceefbcafdd90dda3a8438a00328eb)
 - [b5bce0b4a5795d9ea1b272bac4d073d3d0c50698](https://github.com/magento/magento2/commit/b5bce0b4a5795d9ea1b272bac4d073d3d0c50698)

The fix will be available with the upcoming 2.4.2 release.
